explorer/covidExplorer/CovidParams.ts

```diff
@@ -154,6 +154,17 @@ export class CovidQueryParams {
         this.sizeColumn = params.sizeColumn
         this.colorScale = params.colorScale as colorScaleOption
         this.chartType = params.chartType as ChartTypeName
+
+        // Ensure there is always a metric
+        const hasMetric = [
+            this.cfrMetric,
+            this.casesMetric,
+            this.deathsMetric,
+            this.testsMetric,
+            this.testsPerCaseMetric,
+            this.positiveTestRate
+        ].some(i => i)
+        if (!hasMetric) this.casesMetric = true
```

Comment:
Is there a reason we need to move this?

The idea with the split between `params` and `constrainedParams` is that the former is what the user entered and the latter is a valid state derived from that. Code that reads params should always read from constrainedParams and code that writes should always write to the params. (That should be explicitly coded in, but it's not and this pattern in general is something that comes up a lot so maybe there is some TypeScript voodoo we can do in general).

By moving this up we have no way to distinguish between the case where a user has set "casesMetric" and one where we set it for them. So if you visit live:

 https://ourworldindata.org/coronavirus-data-explorer?zoomToSelection=true&time=2020-03-01..latest&country=MEX~IND~USA~ITA~BRA~GBR~FRA~ESP~PER&interval=smoothed&perCapita=true&smoothing=7&pickerMetric=total_deaths&pickerSort=desc

vs nightingale:

https://nightingale-owid.netlify.app/coronavirus-data-explorer?zoomToSelection=true&time=2020-03-01..latest&country=MEX~IND~USA~ITA~BRA~GBR~FRA~ESP~PER&interval=smoothed&perCapita=true&smoothing=7&pickerMetric=total_deaths&pickerSort=desc

in the latter you'll see `casesMetric=true` get added to the queryString.

This would become more of a problem later on with the switcher explorer.

Reply:
Ah, ok wasn't aware of this distinction. Have reverted. Now, instead of moving this code from constrained to regular, I just moved it earlier in the constrained constructor. The goal is to allow the `available` properties to reflect the default metric we are selecting, rather than rely on the odd state of the unselected metric.

I also added a comment based on the distinction described above. Maybe later we can replace with the Typescript voodoo.